### PR TITLE
Azure/dnsrecordset cleaner

### PR DESCRIPTION
**Pull request by TheoBrigitte**
There is left over DNS record set from CI in our Azure subscription

screenshot_2018-08-23_12-17-58

This PR provide cleaning for DNS record set.

### CODE REVIEW

File: pkg/cleaner/azure/cleaner.go [hunk 1]
```diff
@@ -4,6 +4,7 @@ import (
 	"context"
 	"strings"
 
+	"github.com/Azure/azure-sdk-for-go/services/dns/mgmt/2017-10-01/dns"
```
Comment (MarcelMue) [inline pkg/cleaner/azure/cleaner.go]: Why is this import so much "older" than the others? Or is there simply no newer version?

**Review by MarcelMue — APPROVED** [review]
LGTM

File: pkg/cleaner/azure/cleaner.go (hunk 1 above)
Comment (TheoBrigitte) [reply pkg/cleaner/azure/cleaner.go]: That's the newest beside `dns/mgmt/2018-03-01-preview/dns` where I am not sure what preview stands for, alpha/beta version?
So I rather go with the sable version.

Comment (MarcelMue) [reply pkg/cleaner/azure/cleaner.go]: Ack, makes sense.

**Review by fgimenez — APPROVED** [review]
LGTM

**Pull request merged**